### PR TITLE
Generally relativize text locations for scan results

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

**Comment by sschuberth** [comment]
@fviernau and @mnonnenmacher, I've thought about what we've discussed in today's Kotlin developer meeting about this PR, and while I still agree that it makes sense to ensure relative paths via the individual scanner API, I'd prefer to first merge this as-is (modulo any other review comments) to "collect requirements" (as part of `Scanner.kt`) about that kind of checks / post-processing needs to be done. I have some more smaller refactoring in the pipeline that will probably make easy to move the respective check / post-processing code again.

File: model/src/main/kotlin/TextLocation.kt [hunk 1]
```diff
@@ -90,4 +92,20 @@ data class TextLocation(
...
             else -> min(abs(other.startLine - endLine), abs(startLine - other.endLine))
         }
+
+    /**
+     * Return a [TextLocation] whose path is relative to [basePath], or throw an [IllegalArgumentException] if the paths
+     * have different roots.
+     */
+    fun withRelativePath(basePath: File): TextLocation {
+        val pathAsFile = File(path)
+        if (!pathAsFile.isAbsolute) return this
```
Comment (mnonnenmacher) [inline model/src/main/kotlin/TextLocation.kt:102]: Should this branch also call `invariantSeparatorsPath`?

Comment (sschuberth) [reply model/src/main/kotlin/TextLocation.kt]: Good catch, the corresponding code in `CommandLinePathScannerWrapper` did that as well. Fixed.

File: model/src/main/kotlin/TextLocation.kt [hunk 2]
```diff
@@ -90,4 +92,20 @@ data class TextLocation(
...
         }
+
+    /**
+     * Return a [TextLocation] whose path is relative to [basePath], or throw an [IllegalArgumentException] if the paths
+     * have different roots.
+     */
+    fun withRelativePath(basePath: File): TextLocation {
+        val pathAsFile = File(path)
+
+        val relevantPath = when {
```
Comment (mnonnenmacher) [inline model/src/main/kotlin/TextLocation.kt:103]: Should this be called `relativePath` or `relativeFile` instead of `relevantPath`?

**Review by mnonnenmacher — APPROVED** [review]

**Pull request merged**